lib/ansible/module_utils/network/nxos/config/hsrp_interfaces/hsrp_interfaces.py

```diff
@@ -0,0 +1,244 @@
+        if 'name' not in diff:
+            diff['name'] = want['name']
+
+        replaced_cmds = []
+        if obj_in_have:
+            replaced_cmds = self.del_attribs(diff)
+        if replaced_cmds or merged_cmds:
+            for cmd in set(replaced_cmds).intersection(set(merged_cmds)):
+                merged_cmds.remove(cmd)
+            cmds.extend(replaced_cmds)
+            cmds.extend(merged_cmds)
+        return cmds
+
+    def _state_overridden(self, want, have):
+        """ The command generator when state is overridden
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        cmds = []
+        for h in have:
+            # Check existing states, set to default if not in want or different than want
+            h = flatten_dict(h)
+            obj_in_want = flatten_dict(search_obj_in_list(h['name'], want, 'name'))
+            if obj_in_want:
+                # Let the 'want' loop handle all vals for this interface
+                continue
+            cmds.extend(self.del_attribs(h))
+        for w in want:
+            # Update any want attrs if needed. The overridden state considers
+            # the play as the source of truth for the entire device, therefore
+            # set any unspecified attrs to their default state.
+            w = self.set_none_vals_to_defaults(flatten_dict(w))
+            cmds.extend(self.set_commands(w, have))
+        return cmds
+
+    def _state_merged(self, want, have):
+        """ The command generator when state is merged
+
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        return self.set_commands(want, have)
+
+    def _state_deleted(self, want, have):
+        """ The command generator when state is deleted
+
+        :rtype: A list
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        if not (want or have):
+            return []
+        cmds = []
+        if want:
+            for w in want:
+                obj_in_have = flatten_dict(search_obj_in_list(w['name'], have, 'name'))
+                cmds.extend(self.del_attribs(obj_in_have))
+        else:
+            for h in have:
+                cmds.extend(self.del_attribs(flatten_dict(h)))
+        return cmds
+
+    def del_attribs(self, obj):
+        if not obj or len(obj.keys()) == 1:
+            return []
+        cmds = []
+        if 'bfd' in obj:
+            cmds.append('no hsrp bfd')
+        if cmds:
+            cmds.insert(0, 'interface ' + obj['name'])
+        return cmds
+
+    def set_none_vals_to_defaults(self, want):
+        # Set dict None values to default states
+        if 'bfd' in want and want['bfd'] is None:
+            want['bfd'] = 'disable'
```

Comment:
Why is bfd set to explicitly `disable` if the default state is disable.

Reply:
@chrisvanheuveln Since the default state of bfd itself is `disabled`, the device will set the state to `disabled` in running-config on removal of config  for overridden. Also since `remove_empties` in facts will remove any null valued parameter from the dictionary so in this case `bfd` parameter won't be present. It's the coding logic that needs to modified. Please refer to nxos_vlans module code.

Reply:
The nxos_vlans module is not a valid comparison because it doesn't have any defaults specified in argspec. However, I looked at nxos_interfaces which does (`'enable'` has a default to `True`) and in fact I see the same failure condition for that attribute when I try the same kind of test case (using nxos_interfaces). I'll send you a unit test diff through email that you can try which will demonstrate the failure.

Also, I think I may have confused you so let me clarify. There are two issues but only one is present at a time:

Issue 1. When I set `'default':'disable'` in argspec it breaks the `merged` case. Overridden does work in this case.

Issue 2. When I do NOT set `'default':'disable'` in argspec everything works except for `overridden`, which is why I added the workaround method to handle that.